src/components/NewTodo/NewTodo.tsx

```diff
@@ -0,0 +1,93 @@
+import React, {
+  SyntheticEvent,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import * as todoService from '../../api/todos';
+import { TodoContext } from '../../TodoContext/TodoContext';
+import { ErrorMessages, Todo } from '../../types/Todo';
+
+interface Props {
+  onTodoCreated: (todo: Todo) => void;
+}
+
+export type TodoCreateData = {
+  title: string;
+};
+
+export const NewTodo: React.FC<Props> = ({ onTodoCreated }) => {
+  const { setError, displayError, isLoading, setIsLoading, setTempTodo } =
+    useContext(TodoContext);
+  const [title, setTitle] = useState('');
+
+  const titleField = useRef<HTMLInputElement>(null);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setError(null);
+  };
+
+  const handleSubmit = (event: SyntheticEvent) => {
+    event.preventDefault();
+    if (!title.trim()) {
+      displayError(ErrorMessages.TitleIsEmpty);
+
+      return;
+    }
+
+    setIsLoading(true);
+    todoService
+      .createTodo({
+        userId: todoService.USER_ID,
+        title: { title: title.trim() }.title,
+        completed: false,
+      })
+      .then(newTodo => {
+        onTodoCreated(newTodo);
+        setTitle('');
+      })
+      .catch(() => {
+        displayError(ErrorMessages.AddTodo);
+      })
+      .finally(() => {
+        setIsLoading(false);
+        setTempTodo(null);
+      });
+    setTempTodo({
+      id: 0,
+      userId: todoService.USER_ID,
+      title: { title: title.trim() }.title,
+      completed: false,
+    });
```

Comment:
```suggestion
    const newTodo = {
      userId: todoService.USER_ID,
      title: title.trim(),
      completed: false,
    };
    
    setTempTodo({
      id: 0,
      ...newTodo,
    });
    
    todoService
      .createTodo(newTodo)
      .then(newTodo => {
        onTodoCreated(newTodo);
        setTitle('');
      })
      .catch(() => {
        displayError(ErrorMessages.AddTodo);
      })
      .finally(() => {
        setIsLoading(false);
        setTempTodo(null);
      });
```
`title: { title: title.trim() }.title` why? 🙄